packages/connector-sendgrid-mail/src/index.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorMetadata,
+  EmailSendMessageFunction,
+  ValidateConfig,
+  EmailConnector,
+  GetConnectorConfig,
+} from '@logto/connector-types';
+import { assert, Nullable } from '@silverhand/essentials';
+import got from 'got';
+
+import { defaultMetadata, endpoint } from './constant';
+import {
+  sendGridMailConfigGuard,
+  SendEmailResponse,
+  SendGridMailConfig,
+  EmailData,
+  Personalization,
+  Content,
+  PublicParameters,
+} from './types';
+
+export class SendGridMailConnector implements EmailConnector {
+  public metadata: ConnectorMetadata = defaultMetadata;
+
+  public readonly getConfig: GetConnectorConfig<SendGridMailConfig>;
+
+  constructor(getConnectorConfig: GetConnectorConfig<SendGridMailConfig>) {
+    this.getConfig = getConnectorConfig;
+  }
+
+  public validateConfig: ValidateConfig = async (config: unknown) => {
+    const result = sendGridMailConfigGuard.safeParse(config);
+
+    if (!result.success) {
+      throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error.message);
+    }
+  };
+
+  public sendMessage: EmailSendMessageFunction<Nullable<SendEmailResponse>> = async (
+    address,
+    type,
+    data
+  ) => {
+    const config = await this.getConfig(this.metadata.target, this.metadata.platform);
+    await this.validateConfig(config);
+    const { apiKey, fromEmail, fromName, templates } = config;
+    const template = templates.find((template) => template.usageType === type);
+
+    assert(
+      template,
+      new ConnectorError(
+        ConnectorErrorCodes.TemplateNotFound,
+        `Cannot find template for type: ${type}`
```

Comment:
nit: `cannot find` → `template not found` is more reasonable. (`cannot` seems that there is no way to find)

Reply:
fixed.